### PR TITLE
Lógica da autenticação

### LINKED ISSUE #12: Lógica de autenticação
Criação de uma rota de API para realização do registro e login.
Todas as informações da ferramenta devem estar atribuídas à um usuário.

https://next-auth.js.org/

### PR DESCRIPTION
**Pull request by fpeduu**
Mudanças importantes:
- Estrutura do projeto: A partir de agora, a api do projeto ficará na rota /api/página. Exemplo: 
/api:
    /api/proposal
         /api/proposal/[cpfDevedor]
   /api/auth
- Lógica de autenticação adicionada ao projeto

### CODE REVIEW

File: nextjs_frontend/src/app/api/auth/[...nextauth]/options.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    }),
+    CredentialsProvider({
+      name: "Credentials",
+      credentials: {
+        email: { label: "Email", type: "text", placeholder: "nome@email.com" },
+        password: { label: "Password", type: "password" },
+      },
+      async authorize(credentials) {
+        const response = await fetch(
+          `${process.env.NEXT_PUBLIC_SERVER_URL}/api/auth`,
```
Comment (JDaniloC) [inline nextjs_frontend/src/app/api/auth/[...nextauth]/options.ts]: Here you need to set the serverURL from config/index.ts

File: nextjs_frontend/src/app/api/auth/[...nextauth]/options.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import type { NextAuthOptions } from "next-auth";
+import GoogleProvider from "next-auth/providers/google";
+import CredentialsProvider from "next-auth/providers/credentials";
+
+export const options: NextAuthOptions = {
+  providers: [
+    GoogleProvider({
+      clientId: process.env.GOOGLE_ID as string,
+      clientSecret: process.env.GOOGLE_SECRET as string,
```
Comment (JDaniloC) [inline nextjs_frontend/src/app/api/auth/[...nextauth]/options.ts:10]: These variables aren't specified on README. You need to add to env.example and explain that it's neeeded these variables to work

File: nextjs_frontend/src/app/api/auth/[...nextauth]/options.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+            body: JSON.stringify(credentials),
+          }
+        );
+        const user = await response.json();
+
+        if (user) {
+          return user;
+        } else {
+          return null;
+        }
```
Comment (JDaniloC) [inline nextjs_frontend/src/app/api/auth/[...nextauth]/options.ts]: If you don't returns nothing, will return null by default

File: nextjs_frontend/src/middleware.ts
Comment (JDaniloC) [inline nextjs_frontend/src/middleware.ts:1]: You created a middleware that does nothing. It's to privacy routes in the feature?

**Review by JDaniloC — CHANGES_REQUESTED** [review]
Explain it

File: nextjs_frontend/src/app/api/auth/[...nextauth]/options.ts (hunk 1 above)
Comment (fpeduu) [reply nextjs_frontend/src/app/api/auth/[...nextauth]/options.ts]: 😭

File: nextjs_frontend/src/app/api/auth/[...nextauth]/options.ts (hunk 2 above)
Comment (fpeduu) [reply nextjs_frontend/src/app/api/auth/[...nextauth]/options.ts:10]: ok!

File: nextjs_frontend/src/middleware.ts
Comment (fpeduu) [reply nextjs_frontend/src/middleware.ts:1]: it does, it protects any route the matcher catches.

if i don't set any matcher, it'll just make the whole project protected by authentication.

**Review by Vinicius-resende-cin — APPROVED** [review]

**Review by karensamara — APPROVED** [review]

**Review by JDaniloC — APPROVED** [review]
Ok!

**Pull request merged**